piptools/writer.py

```diff
@@ -69,13 +70,22 @@ def _iter_lines(self, results, reverse_dependencies, primary_packages):
             yield line
 
         if unsafe_packages:
-            yield ''
-            yield comment('# The following packages are commented out because they are')
-            yield comment('# considered to be unsafe in a requirements file:')
-
-            for ireq in unsafe_packages:
-                line = self._format_requirement(ireq, reverse_dependencies, primary_packages, include_specifier=False)
-                yield comment('# ' + line)
+            if self.unsafe:
+                yield ''
+                yield comment('# The following packages are included because pip-tools was invoked with --unsafe')
+                yield comment('# However, they may be considered unsafe in a requirements file:')
+
+                for ireq in unsafe_packages:
+                    line = self._format_requirement(ireq, reverse_dependencies, primary_packages)
+                    yield line
```

Comment:
I like the idea for this flag. Rather than putting in an explicit comment here and repeat some of the code, I'd be fine simply setting the `unsafe_packages` set to the empty set (on line 62) if this flag is active. This will not make setuptools stand out from the rest (but that's the intention if this flag is used anyway).